src/parser.jl

```diff
@@ -2617,6 +2617,9 @@ function parse_array(ps::ParseState, mark, closer, end_is_symbol)
     # For an excellent overview of Pratt parsing, see
     # https://matklad.github.io/2020/04/13/simple-but-powerful-pratt-parsing.html
     (dim, binding_power) = parse_array_separator(ps, array_order)
+    if binding_power == typemin(Int)
+        return (K"hcat", EMPTY_FLAGS)
```

Comment:
Let's consume `]` if it's present.

```suggestion
        bump_closing_token(ps, closer)
        return (K"hcat", EMPTY_FLAGS)
```